jenkins-master/Dockerfile

```diff
@@ -31,3 +31,6 @@ USER jenkins
 ENV JAVA_OPTS $JAVA_OPTS -Djenkins.install.runSetupWizard=false -Dhudson.model.DirectoryBrowserSupport.CSP=
 
 EXPOSE 8080 8443
+
+# Files required for DockerHub Autotest
+COPY tests/* /var/jenkins_home/
```

Comment:
Do we really want to copy that into /var/jenkins_home. That will then also be in the volumes of users.

Reply:
True, it was the most simple way to avoid permission issues, but we can also create another directory and allow writing for the `jenkins` user there.

Reply:
I'm also not happy with putting test coding into the productive image. There are good reasons for keeping test coding and productive coding clearly separated. If that is required at all we should put it into a folder so that it is clear that it is about tests, e.g. `/tmp/test`, rather than polluting `$JENKINS_HOME`. But would be even better if there is no test coding inside the "payload". But I'm not sure if that is possible here.

Reply:
> But would be even better if there is no test coding inside the "payload"

As of my current understanding, this would only be possible if we built a new image for the test, which then would be the unit under test, so not the original image anymore which defeats the purpose. I moved the files to a dedicated directory.